mass/adapters/inbound/fastapi_/routes.py

```diff
@@ -15,6 +15,66 @@
 #
 
 """Contains API endpoints"""
-from fastapi import APIRouter
+from typing import Union
+
+from dependency_injector.wiring import Provide, inject
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+
+from mass.config import Config
+from mass.container import Container
+from mass.core import models
+from mass.ports.inbound.query_handler import (
+    ClassNotConfiguredError,
+    QueryHandlerPort,
+    SearchError,
+)
 
 router = APIRouter()
+
+
+# GET /rpc/search-options
+@router.get(
+    path="/rpc/search-options",
+    summary="Retrieve all configured resource classes and facetable properties",
+    response_model=dict[str, models.SearchableClass],
+)
+@inject
+async def search_options(
+    config: Config = Depends(Provide[Container.config]),
+) -> dict[str, models.SearchableClass]:
+    """Returns the configured searchable classes"""
+
+    return config.searchable_classes
+
+
+# POST /rpc/search
+@router.post(
+    path="/rpc/search",
+    summary="Perform a search using query string and filter parameters",
+    response_model=models.QueryResults,
+)
+@inject
+async def search(
+    parameters: models.SearchParameters,
+    query_handler: QueryHandlerPort = Depends(Provide[Container.query_handler]),
+) -> Union[models.QueryResults, None]:
+    """Perform search query"""
+    try:
+        results = await query_handler.handle_query(
+            class_name=parameters.class_name,
+            query=parameters.query,
+            filters=parameters.filters,
+            skip=parameters.skip,
+            limit=parameters.limit,
+        )
+    except ClassNotConfiguredError as err:
+        raise HTTPException(
+            status_code=404, detail="The specified class name is invalid"
```

Comment:
404 is rather when the url cannot be found. However, here the payload is malformed. In that situation a 422 might be better:

```suggestion
            status_code=422, detail="The specified class name is invalid"
```

Reply:
That makes sense. I updated it in [072b380](https://github.com/ghga-de/mass/pull/7/commits/072b3800ef9b2ac017a691a68f1df8e15ea8d287).